tests/tests/test-fees/test-fee-multiplier.ts

```diff
@@ -160,3 +156,239 @@ describeDevMoonbeam("Max Fee Multiplier - initial value", (context) => {
+    const initialValue = await context.polkadotApi.query.transactionPayment.nextFeeMultiplier();
+    await context.polkadotApi.tx.balances
+      .transfer(BALTATHAR_ADDRESS, 1_000_000_000_000_000_000n)
+      .signAndSend(alith, { nonce: -1 });
+    await context.polkadotApi.tx.sudo
+      .sudo(context.polkadotApi.tx.system.fillBlock(TARGET_FILL_AMOUNT))
+      .signAndSend(alith, { nonce: -1 });
+    await context.createBlock();
+
+    const postValue = await context.polkadotApi.query.transactionPayment.nextFeeMultiplier();
+    expect(initialValue.lt(postValue), "Fee multiplier not increased when above ideal fill ratio")
+      .to.be.true;
+  });
+
+  it("should not increase fees with xcm activity", async () => {
+    const transferCallEncoded = context.polkadotApi.tx.balances
+      .transfer(random.address, transferredBalance / 10n)
+      .method.toHex();
+
+    const initialValue = await context.polkadotApi.query.transactionPayment.nextFeeMultiplier();
+    const initialBalance = (await context.polkadotApi.query.system.account(random.address)).data
+      .free;
+
+    await context.polkadotApi.tx.sudo
+      .sudo(context.polkadotApi.tx.system.fillBlock(TARGET_FILL_AMOUNT))
+      .signAndSend(alith, { nonce: -1 });
+    const xcmMessage = new XcmFragment({
+      fees: {
+        multilocation: [
+          {
+            parents: 0,
+            interior: {
+              X1: { PalletInstance: balancesPalletIndex },
+            },
+          },
+        ],
+        fungible: transferredBalance / 3n,
+      },
+      weight_limit: new BN(4000000000),
+      descend_origin: sendingAddress,
+    })
+      .descend_origin()
+      .withdraw_asset()
+      .buy_execution()
+      .push_any({
+        Transact: {
+          originType: "SovereignAccount",
+          requireWeightAtMost: new BN(1000000000),
+          call: {
+            encoded: transferCallEncoded,
+          },
+        },
+      })
+      .push_any({
+        Transact: {
+          originType: "SovereignAccount",
+          requireWeightAtMost: new BN(1000000000),
+          call: {
+            encoded: transferCallEncoded,
+          },
+        },
+      })
+      .push_any({
+        Transact: {
+          originType: "SovereignAccount",
+          requireWeightAtMost: new BN(1000000000),
+          call: {
+            encoded: transferCallEncoded,
+          },
+        },
+      })
+      .as_v2();
+
+    await injectHrmpMessageAndSeal(context, 1, {
+      type: "XcmVersionedXcm",
+      payload: xcmMessage,
+    } as RawXcmMessage);
+
+    const postValue = await context.polkadotApi.query.transactionPayment.nextFeeMultiplier();
+    const postBalance = (await context.polkadotApi.query.system.account(random.address)).data.free;
+
+    expect(initialBalance.lt(postBalance), "Expected balances not updated").to.be.true;
+    expect(initialValue.eq(postValue), "Fee Multiplier has changed between blocks").to.be.true;
```

Comment:
I think some of these tests (like this one) could silently pass if they failed to create a block. I'm probably being too paranoid, though, any defense against this would just be bloat...

Reply:
Good point, have added another expect for block height to guard against this .